### PR TITLE
Moar dataclasses, part 3

### PR DESCRIPTION
**Pull request by andlaus**
the third batch of dataclass conversions. this one converts quite a few classes, but they are all quite simple...

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/diaglayercontainer.py [hunk 1]
```diff
@@ -71,26 +56,26 @@ def from_et(et_element: ElementTree.Element) -> "DiagLayerContainer":
...
-        ]
-        base_variants = [
+        ])
+        base_variants = NamedItemList(short_name_as_id, [
             DiagLayer.from_et(dl_element, doc_frags)
             for dl_element in et_element.iterfind("BASE-VARIANTS/BASE-VARIANT")
-        ]
-        ecu_variants = [
+        ])
+        ecu_variants = NamedItemList(short_name_as_id, [
```
Comment (kayoub5) [inline odxtools/diaglayercontainer.py:75]: is short_name_as_id actually needed anymore?

File: odxtools/request.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 from .basicstructure import BasicStructure
 
 
+@dataclass
 class Request(BasicStructure):
 
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-
     def __repr__(self) -> str:
```
Comment (kayoub5) [inline odxtools/request.py:10]: useless repr, use dataclass auto implementation

File: odxtools/diaglayercontainer.py (hunk 1 above)
Comment (andlaus) [reply odxtools/diaglayercontainer.py:75]: yes: without it, `NamedItemList` does not know how to map an item to a name. also, ODX short names can start with a number, so we cannot simply pass through the short name a la `lambda x: x.short_name`...

Comment (kayoub5) [reply odxtools/diaglayercontainer.py:75]: do you have actually a case where NamedItemList have another id mapper other than short_name_as_id?

File: odxtools/request.py (hunk 2 above)
Comment (andlaus) [reply odxtools/request.py]: right. fixed: 544249a51be18d56a3ee9f44bf40668becffc7ef

File: odxtools/diaglayercontainer.py (hunk 1 above)
Comment (andlaus) [reply odxtools/diaglayercontainer.py:75]: not within odxtools. (I think that this class ought to be implemented generically, IMO). To make its usage a bit more convenient, we could introduce a `ShortNameItemList` class for this purpose, though...

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**